token/link/link.go

```diff
@@ -53,7 +54,17 @@ type LinkConfig interface {
 
 // OauthProviderFactory represents oauth provider factory
 type OauthProviderFactory interface {
-	NewOauthProvider(ctx context.Context, req *goa.RequestData, config LinkConfig, forResource string) (ProviderConfig, error)
+	NewOauthProvider(ctx context.Context, req *goa.RequestData, forResource string) (ProviderConfig, error)
+}
+
+// NewOauthProviderFactory returns the default Oauth provider factory.
+func NewOauthProviderFactory(config LinkConfig, db application.DB) OauthProviderFactory {
```

Comment:
db is not really used in the provider factory implemented by link service, is it? I actually saved a few lines of code by having one implementation (link service) for both interfaces. But maybe it will be better to split them to two structs if you need to use them outside of the link service. It could be less error pron.

Reply:
I modified the link service a bit.
I moved out the NewOAuthProviderFactory(..) outside linkservice
4a1292652d4dd64408a6162a9b621b2108fa18eb 
Also, removed `NewLinkService()` and kept just one `NewLinkServiceWithFactory()` method. ( Should I bring back the `NewLinkService(..) method and create the factory provider inside it ? A bit unsure if 'aways-explicit' is OK )

So the provider factory is now being consumed by `LinkService` too.

( btw ignore the code I commented out, I've put it back ba6db10a91766a30e1f35e62290009e3c96f9672 )